akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ReceiveBuilder.scala

```diff
@@ -5,37 +5,40 @@
 package akka.actor.typed.javadsl
 
 import scala.annotation.tailrec
-import akka.japi.function.{ Creator, Function, Predicate }
+import akka.japi.function.Creator
+import akka.japi.function.{ Function ⇒ JFunction }
+import akka.japi.function.{ Predicate ⇒ JPredicate }
 import akka.actor.typed.{ Behavior, Signal }
 import akka.annotation.InternalApi
+import akka.util.OptionVal
 
 /**
- * Used when implementing [[AbstractBehavior]].
+ * Mutable builder used when implementing [[AbstractBehavior]].
  *
  * When handling a message or signal, this [[Behavior]] will consider all handlers in the order they were added,
  * looking for the first handler for which both the type and the (optional) predicate match.
  *
  * @tparam T the common superclass of all supported messages.
  */
-class ReceiveBuilder[T] private (
-  private val messageHandlers: List[ReceiveBuilder.Case[T, T]],
-  private val signalHandlers:  List[ReceiveBuilder.Case[T, Signal]]
+final class ReceiveBuilder[T] private (
+  private var messageHandlers: List[ReceiveBuilder.Case[T, T]],
+  private var signalHandlers:  List[ReceiveBuilder.Case[T, Signal]]
 ) {
```

Comment:
do you thing we should add `orElse`? command and event builders have that
(separate PR)

Reply:
I think it is less useful here than in typed persistence. Could be nice for consistency though.

Reply:
Coming late to the party, but...

I don't think it's less useful. You can define different `ReceiveBuilder` and compose then differently depending on each state/behavior your are. That was a common pattern in untyped, no? Combining different `Receive` PF. 

Or am I missing something here?